agent/api/ecsclient/client.go

```diff
@@ -191,11 +201,40 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 
 	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
 	if err != nil {
-		log.Error("Could not register", "err", err)
+		log.Error("Could not register", "error", err)
 		return "", err
 	}
 	log.Info("Registered!")
-	return *resp.ContainerInstance.ContainerInstanceArn, nil
+	err = ValidateRegisteredAttributes(&registerRequest.Attributes, &resp.ContainerInstance.Attributes)
+	return *resp.ContainerInstance.ContainerInstanceArn, err
+}
+
+func attributesToMap(attributes *[]*ecs.Attribute) map[string]string {
```

Comment:
You shouldn't need to pass a `*[]*ecs.Attribute` to this; a `[]*ecs.Attribute` should be sufficient.  Slices are already pointers.